changelogs/CHANGELOG-v1.20.0-beta.1.md

```diff
@@ -0,0 +1,216 @@
+
+## Gateway API: implement PathPrefix matching
+
+Contour now implements Gateway API v1alpha2's "path prefix" matching for `HTTPRoutes`.
+This is now the only native form of prefix matching supported by Gateway API, and is a change from v1alpha1.
+Path prefix matching means that the prefix specified in an `HTTPRoute` rule must match entire segments of a request's path in order to match it, rather than just be a string prefix.
+For example, the prefix `/foo` would match a request for the path `/foo/bar` but not `/foobar`.
+For more information, see the [Gateway API documentation](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.PathMatchType).
+
+(#4119, @skriss)
+
+## Gateway API: support ReferencePolicy
+
+Contour now supports the `ReferencePolicy` CRD in Gateway API v1alpha2.
+`ReferencePolicy` enables certain cross-namespace references to be allowed in Gateway API.
+The primary use case is to enable routes (e.g. `HTTPRoutes`, `TLSRoutes`) to reference backend `Services` in different namespaces.
+When Contour processes a route that references a service in a different namespace, it will check for a `ReferencePolicy` that applies to the route and service, and if one exists, it will allow the reference.
+
+(#4138, @skriss)
+
+## Gateway API: set Gateway Listener status fields
+
+Contour now sets the `.status.listeners.supportedKinds` and `.status.listeners.attachedRoutes` fields on Gateways for Gateway API.
+The first describes the list of route groups/kinds that the listener supports, and the second captures the total number of routes that are successfully attached to the listener.
+
+(#4160, @skriss)
+
+## Set Gateway listener conditions
+
+Contour now sets various Gateway listener conditions as it processes them, including the "Ready", "Detached", and "ResolvedRefs" condition types, to provide more visibility to the user as to whether their listeners are defined correctly or not.
+
+(#4186, @skriss)
+
+## Default status on HTTPProxy resources
+
+When a new HTTPProxy is created, if Contour isn't yet running or
+functioning properly, then no status is set on the resource. 
+Defaults of "NotReconciled/Waiting for controller" are now applied 
+to any new object until an instance of Contour accepts the
+object and updates the status.
+
+(#4133, @stevesloka)
+
+## Source IP hash based load balancing
+
+Contour users can now configure their load balancing policies on `HTTPProxy` resources to hash the source IP of a client to ensure consistent routing to a backend service instance. Using this feature combined with header value hashing can implement advanced request routing and session affinity. Note that if you are using a load balancer to front your Envoy deployment, you will need to ensure it preserves client source IP addresses to ensure this feature is effective.
+
+See [this page](https://projectcontour.io/docs/v1.20.0/config/request-routing/#load-balancing-strategy) for more details on this feature.
```

Comment:
ah, didnt anticipate a beta when writing this one, might need to adjust/remove this? i guess we won't have beta versioned docs

Reply:
could just adjust this instance to `main` instead of a version

Reply:
Good catch, I redirected to `main` docs.